src/CodeGenerator/AbstractInjector.php

```diff
@@ -13,10 +13,10 @@
  */
 abstract class AbstractInjector implements InjectorInterface
 {
-    /** @var string[]|FactoryInterface[] */
+    /** @var array<string, class-string<FactoryInterface>|FactoryInterface> */
```

Comment:
*maybe* bump minimum requirement of `laminas-servicemanager`.
I've added plenty of psalm types there which can be imported here for re-usage.

Reply:
laminas-servicemanager is a dev-dependency and it's not ment to be a requirement. 
Importing types from this project would require to make it a runtime dependency just to pass static analysis, which is too much imho.

Reply:
This is tested against a recent and locked servicemanager: all good